src/Term/TermFallback.php

```diff
@@ -46,6 +46,18 @@ public function __construct( $requestedLanguageCode, $text, $actualLanguageCode,
 			throw new InvalidArgumentException( '$sourceLanguageCode must be a non-empty string or null' );
 		}
 
+		if ( $actualLanguageCode === $sourceLanguageCode ) {
+			throw new InvalidArgumentException(
+				'$actualLanguageCode and $sourceLanguageCode must be different'
+			);
+		}
+
+		if ( $sourceLanguageCode === null && $actualLanguageCode !== $requestedLanguageCode ) {
```

Comment:
I think this is wrong. If "requested" is, for example, `de-at`, "actual" can be `de` with no "source" (no "source" that's different from "actual").